docs/tutorial/t-create-client-credentials.md

```diff
@@ -0,0 +1,399 @@
+
+```bash
+juju ssh kafka/leader "
+echo 'LOG: Creating ACLs for SASL user'
+sudo charmed-kafka.acls --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client.properties \
+  --add --allow-principal User:$KAFKA_CLIENT_SASL_USER \
+  --operation READ --operation DESCRIBE --group='*'
+
+sudo charmed-kafka.acls --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client.properties \
+  --add --allow-principal User:$KAFKA_CLIENT_SASL_USER \
+  --operation READ --operation DESCRIBE --operation CREATE --operation WRITE --operation DELETE --operation ALTER --operation ALTERCONFIGS --topic='*'
+
+sudo charmed-kafka.acls --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client.properties \
+  --add --allow-principal User:$KAFKA_CLIENT_SASL_USER \
+  --operation DESCRIBE --operation WRITE --transactional-id '*'
+"
+```
+
+```bash
+juju ssh kafka/leader "
+echo 'LOG: Creating ACLs for MTLS user'
+sudo charmed-kafka.acls --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client.properties \
+  --add --allow-principal User:$KAFKA_CLIENT_MTLS_CN \
+  --operation READ --operation DESCRIBE --group='*'
+
+sudo charmed-kafka.acls --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client.properties \
+  --add --allow-principal User:$KAFKA_CLIENT_MTLS_CN \
+  --operation READ --operation DESCRIBE --topic='*'
+
+sudo charmed-kafka.acls --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client.properties \
+  --add --allow-principal User:$KAFKA_CLIENT_MTLS_CN \
+  --operation DESCRIBE --transactional-id '*'
+"
+```
+
+> Read more at https://docs.confluent.io/platform/current/kafka/authorization.html#operations
+
+## Testing access
+
+```bash
+# ---------- Testing Kafka Access
+KAFKA_TEST_TOPIC=TEST-TOPIC-A
+
+sudo cp client.truststore.jks $SNAP_KAFKA_PATH/
+sudo cp client.keystore.p12 $SNAP_KAFKA_PATH/
+sudo cp client-mtls.properties $SNAP_KAFKA_PATH/
+sudo cp client-sasl.properties $SNAP_KAFKA_PATH/
+
+sudo chown snap_daemon:root $SNAP_KAFKA_PATH/client-mtls.properties
+sudo chown snap_daemon:root $SNAP_KAFKA_PATH/client-sasl.properties
+sudo chown snap_daemon:root $SNAP_KAFKA_PATH/client.keystore.p12
+sudo chown snap_daemon:root $SNAP_KAFKA_PATH/client.truststore.jks
+
+# mTLS
+sudo charmed-kafka.topics --bootstrap-server $KAFKA_SERVERS_MTLS --command-config $SNAP_KAFKA_PATH/client-mtls.properties \
+  --create --topic $KAFKA_TEST_TOPIC
+sudo charmed-kafka.topics --list --bootstrap-server $KAFKA_SERVERS_MTLS --command-config $SNAP_KAFKA_PATH/client-mtls.properties
+
+# SASL_SSL
+sudo charmed-kafka.topics --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client-sasl.properties \
+  --create --topic $KAFKA_TEST_TOPIC
+sudo charmed-kafka.topics --list --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client-sasl.properties
+```
```

Comment:
```suggestion
```bash
# ---------- Testing Kafka Access
KAFKA_TEST_TOPIC=EXAMPLE-TOPIC

# copying the files to a path readable by the `charmed-kafka` snap commands
sudo cp client.truststore.jks $SNAP_KAFKA_PATH/
sudo cp client.keystore.p12 $SNAP_KAFKA_PATH/
sudo cp client-mtls.properties $SNAP_KAFKA_PATH/
sudo cp client-sasl.properties $SNAP_KAFKA_PATH/

# confirming correct file permissions to be readable by the snap
sudo chown snap_daemon:root $SNAP_KAFKA_PATH/client-mtls.properties
sudo chown snap_daemon:root $SNAP_KAFKA_PATH/client-sasl.properties
sudo chown snap_daemon:root $SNAP_KAFKA_PATH/client.keystore.p12
sudo chown snap_daemon:root $SNAP_KAFKA_PATH/client.truststore.jks

# mTLS
sudo charmed-kafka.topics --bootstrap-server $KAFKA_SERVERS_MTLS --command-config $SNAP_KAFKA_PATH/client-mtls.properties \
  --create --topic $KAFKA_TEST_TOPIC
sudo charmed-kafka.topics --list --bootstrap-server $KAFKA_SERVERS_MTLS --command-config $SNAP_KAFKA_PATH/client-mtls.properties

# SASL_SSL
sudo charmed-kafka.topics --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client-sasl.properties \
  --create --topic $KAFKA_TEST_TOPIC
sudo charmed-kafka.topics --list --bootstrap-server $KAFKA_SERVERS_SASL --command-config $SNAP_KAFKA_PATH/client-sasl.properties
```
```